src/raft.rs

```diff
@@ -717,9 +730,11 @@ impl<T: Storage> Raft<T> {
 
     /// Converts this node to a follower.
     pub fn become_follower(&mut self, term: u64, leader_id: u64) {
+        let pending_request_snapshot = self.pending_request_snapshot;
```

Comment:
These 2 line seems useless?

Reply:
Because reset will reset the pending_request_snapshot, so we should save it first.